libsolidity/analysis/TypeChecker.cpp

```diff
@@ -2033,18 +2033,29 @@ void TypeChecker::typeCheckBytesConcatFunction(
 	typeCheckFunctionGeneralChecks(_functionCall, _functionType);
 
 	for (shared_ptr<Expression const> const& argument: _functionCall.arguments())
-		if (
-			Type const* argumentType = type(*argument);
+	{
+		Type const* argumentType = type(*argument);
+		bool notConvertibleToBytes = argumentType &&
```

Comment:
Can it be null?

Reply:
Good point. I think it can't. I'll change that.

Reply:
Done.

I thought that this check was here before but I just misinterpreted the `if` with variable initialization in it - the initialization does not automatically check if the variable does not evaluate to `false`. I really do not like this `if` syntax. I just keep reading it wrong.